### DISCUSSION TITLE
How can I make the action button in my form toggle the visibility of another section?

### DISCUSSION THREAD

**Discussion by bsamworthdev** [filamentphp/filament#9593] [Help] [answered]
I've tried doing this but the visibility doesn't change.

private $showPreviousAddress = true;
...
Section::make('Person')
                ->schema([
                    TextInput::make('residentialaddress1')->label('Name'),
                    Actions::make([
                            Action::make('Show/Hide Details')
                                ->action(function () {
                                    $this->showHideDetails= !$this->showHideDetails;
                                })
                        ]),
                ]),
            Section::make('Person Details')
                ->visible($this->showHideDetails)
                ->schema([
                    TextInput::make('prevresidentialaddress1')->label('Address Line 1'),
                    TextInput::make('prevresidentialpostcode')->label('Postcode')
                ]),

**Comment by leandrocfe** [comment, marked as answer]
You could use a toggle

```php
Section::make('Person')
    ->schema([
        TextInput::make('residentialaddress1')->label('Name'),
        Toggle::make('show_details')
            ->live(),
    ]),
Section::make('Person Details')
    ->visible(fn (Get $get): bool => $get('show_details'))
    ->schema([
        TextInput::make('prevresidentialaddress1')->label('Address Line 1'),
        TextInput::make('prevresidentialpostcode')->label('Postcode'),
    ])
```

**Reply by bsamworthdev** [reply]
This approach assumes there is a show_details field in the database. What if I just want it to toggle the visibility and not store it anywhere?

**Reply by bsamworthdev** [reply]
I got it working by using:
                                ->requiredWith('show_details')
                                ->reactive()

**Reply by leandrocfe** [reply]
> This approach assumes there is a show_details field in the database. What if I just want it to toggle the visibility and not store it anywhere?

You can use `->dehydrated(false)`
